File: src/web-ui/src/infrastructure/api/service-api/GitAgentAPI.ts


import { api } from './ApiClient';
import { createTauriCommandError } from '../errors/TauriCommandError';




export type CommitType = 
  | 'Feat' 
  | 'Fix' 
  | 'Docs' 
  | 'Style' 
  | 'Refactor' 
  | 'Perf' 
  | 'Test' 
  | 'Chore' 
  | 'CI' 
  | 'Revert';


export type CommitFormat = 'Conventional' | 'Angular' | 'Simple' | 'Custom';


export type Language = 'Chinese' | 'English';


export type FileChangeType = 'Added' | 'Modified' | 'Deleted' | 'Renamed';


export type ChangePattern = 
  | 'FeatureAddition'
  | 'BugFix'
  | 'Refactoring'
  | 'ConfigChange'
  | 'DependencyUpdate'
  | 'DocumentationUpdate'
  | 'TestUpdate'
  | 'StyleChange';



export interface GenerateCommitMessageRequest {
  repoPath: string;
  options?: CommitMessageOptions;
}

export interface QuickCommitMessageRequest {
  repoPath: string;
}

export interface PreviewCommitMessageRequest {
  repoPath: string;
}



export interface CommitMessageOptions {
  format?: CommitFormat;
  includeFiles?: boolean;
  maxTitleLength?: number;
  includeBody?: boolean;
  language?: Language;
}



export interface CommitMessage {
  title: string;
  body?: string;
  footer?: string;
  fullMessage: string;
  commitType: CommitType;
  scope?: string;
  confidence: number;
  changesSummary: ChangesSummary;
}

export interface ChangesSummary {
  totalAdditions: number;
  totalDeletions: number;
  filesChanged: number;
  fileChanges: FileChange[];
  affectedModules: string[];
  changePatterns: ChangePattern[];
}

export interface FileChange {
  path: string;
  changeType: FileChangeType;
  additions: number;
  deletions: number;
  fileType: string;
}

export interface PreviewCommitMessageResponse {
  title: string;
  commitType: string;
  scope?: string;
  confidence: number;
  filesChanged: number;
  additions: number;
  deletions: number;
}



export class GitAgentAPI {
   
  async generateCommitMessage(request: GenerateCommitMessageRequest): Promise<CommitMessage> {
    try {
      return await api.invoke<CommitMessage>('generate_commit_message', { request });
    } catch (error) {
      throw createTauriCommandError('generate_commit_message', error, request);
    }
  }

   
  async quickCommitMessage(request: QuickCommitMessageRequest): Promise<CommitMessage> {
    try {
      return await api.invoke<CommitMessage>('quick_commit_message', { request });
    } catch (error) {
      throw createTauriCommandError('quick_commit_message', error, request);
    }
  }

   
  async quickCommit(repoPath: string): Promise<CommitMessage> {
    return this.quickCommitMessage({ repoPath });
  }

   
  async previewCommitMessage(request: PreviewCommitMessageRequest): Promise<PreviewCommitMessageResponse> {
    try {
      return await api.invoke<PreviewCommitMessageResponse>('preview_commit_message', { request });
    } catch (error) {
      throw createTauriCommandError('preview_commit_message', error, request);
    }
  }

   
  async previewCommit(repoPath: string): Promise<PreviewCommitMessageResponse> {
    return this.previewCommitMessage({ repoPath });
  }
}



export const gitAgentAPI = new GitAgentAPI();



export default gitAgentAPI;

File: src/apps/desktop/src/api/git_agent_api.rs
//! Git Agent API - Provides Tauri command interface for Git Function Agent

use log::error;
use bitfun_core::function_agents::{
    GitFunctionAgent,
    CommitMessage,
    CommitMessageOptions,
};
use crate::api::app_state::AppState;
use serde::{Deserialize, Serialize};
use std::path::Path;
use tauri::State;

#[derive(Debug, Clone, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct GenerateCommitMessageRequest {
    pub repo_path: String,
    pub options: Option<CommitMessageOptions>,
}

#[derive(Debug, Clone, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct QuickCommitMessageRequest {
    pub repo_path: String,
}

#[derive(Debug, Clone, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct PreviewCommitMessageRequest {
    pub repo_path: String,
}

#[derive(Debug, Clone, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct PreviewCommitMessageResponse {
    pub title: String,
    pub commit_type: String,
    pub scope: Option<String>,
    pub confidence: f32,
    pub files_changed: u32,
    pub additions: u32,
    pub deletions: u32,
}

#[tauri::command]
pub async fn generate_commit_message(
    app_state: State<'_, AppState>,
    request: GenerateCommitMessageRequest,
) -> Result<CommitMessage, String> {
    let factory = app_state.ai_client_factory.clone();
    let agent = GitFunctionAgent::new(factory);
    let opts = request.options.unwrap_or_default();
    
    agent
        .generate_commit_message(Path::new(&request.repo_path), opts)
        .await
        .map_err(|e| e.to_string())
}

#[tauri::command]
pub async fn quick_commit_message(
    app_state: State<'_, AppState>,
    request: QuickCommitMessageRequest,
) -> Result<CommitMessage, String> {
    let factory = app_state.ai_client_factory.clone();
    let agent = GitFunctionAgent::new(factory);
    
    agent
        .quick_commit_message(Path::new(&request.repo_path))
        .await
        .map_err(|e| {
            error!("Failed to generate quick commit message: repo_path={}, error={}", request.repo_path, e);
            e.to_string()
        })
}

#[tauri::command]
pub async fn preview_commit_message(
    app_state: State<'_, AppState>,
    request: PreviewCommitMessageRequest,
) -> Result<PreviewCommitMessageResponse, String> {
    let factory = app_state.ai_client_factory.clone();
    let agent = GitFunctionAgent::new(factory);
    
    let message = agent
        .quick_commit_message(Path::new(&request.repo_path))
        .await
        .map_err(|e| e.to_string())?;
    
    Ok(PreviewCommitMessageResponse {
        title: message.title,
        commit_type: format!("{:?}", message.commit_type),
        scope: message.scope,
        confidence: message.confidence,
        files_changed: message.changes_summary.files_changed,
        additions: message.changes_summary.total_additions,
        deletions: message.changes_summary.total_deletions,
    })
}
